boards: nrf52840dongle: update flashing instructions

mrodgers-witekio opened the pull request:
I noticed when trying to flash an `nrf52840dongle` via the built-in bootloader that the tool linked from the board index file is deprecated, and has been replaced with a newer tool.

Update the flashing instructions to link to the newer tool, and update the command line examples accordingly

mrodgers-witekio committed:
boards: nrf52840dongle: update flashing instructions

Update flashing instructions using built-in bootloader to use latest
version of tool for packaging and uploading images.

Signed-off-by: Matt Rodgers <mrodgers@witekio.com>

kartben reviewed (dismissed):
I'm guessing other boards might need a similar update?

mrodgers-witekio commented:
> I'm guessing other boards might need a similar update?

I hadn't considered this, but a quick `grep` shows that `boards/ezurio/bl654_usb/doc/bl654_usb.rst` uses similar flashing instructions and could probably do with an update. Happy to add this in as well.

mrodgers-witekio committed:
boards: bl654_usb: update flashing instructions

    Update flashing instructions using built-in bootloader to use latest
    version of tool for packaging and uploading images.

Signed-off-by: Matt Rodgers <mrodgers@witekio.com>

anangl commented on boards/nordic/nrf52840dongle/doc/index.rst:
```diff
@@ -198,16 +204,19 @@ to the zephyr repository on your computer.
+      nrfutil nrf5sdk-tools pkg generate \
+               --hw-version 52 \
+               --sd-req=0x00 \
+               --application build/zephyr/zephyr.hex \
```
```suggestion
               --application build/mcuboot/zephyr/zephyr.hex \
```

carlescufi reviewed (approved):
Thanks for this!
